src/adjoint/adjointDebug.F90

```diff
@@ -0,0 +1,736 @@
+! This module is used for debugging and testing only
+
+module adjointDebug
+
+contains
+
+#ifndef USE_COMPLEX
+
+    subroutine computeMatrixFreeProductFwdFD(xvdot, extradot, wdot, bcDataValuesdot,&
```

Comment:
Overall, it feels that this shall be located in the same file with computeMatrixFreeProductFwd and computeMatrixFreeProductBwd similar with master to master_d and master_b because the function is very similar with computeMatrixFreeProductFwd.

Reply:
I am fine with having it here as long as all functions here are for debugging and never used for actual production runs. Plus this is probably easier to include these in a complex compile, I think the other adjoint stuff is not included in complex build.